@xen-orchestra/lite/src/libs/xapi.ts

```diff
@@ -3,18 +3,41 @@ import { EventEmitter } from 'events'
 import { Map } from 'immutable'
 
 export interface XapiObject {
+  $pool: Pool
+  $ref: string
   $type: keyof types
   $id: string
 }
 
 // Dictionary of XAPI types and their corresponding TypeScript types
 interface types {
+  network: Network
+  PIF: Pif
+  pool: Pool
   VM: Vm
   host: Host
 }
 
 // Types ---
 
+export interface Network extends XapiObject {
+  name_label: string
+}
+
+export interface Pif extends XapiObject {
+  device: string
+  DNS: string
+  gateway: string
+  IP: string
+  management: boolean
+  network: string
+}
+
+type _Pool = Omit<XapiObject, '$pool'>
```

Comment:
Why do you want to omit `$pool`?

Reply:
There is no point having a pool object in a pool object

Reply:
- it can be useful when you have an unknown XAPI object and want to get its pool
- xen-api provides that property anyway so there's no point in removing it from the type either